networks/rpc/server.go

```diff
@@ -145,6 +146,17 @@ func (s *Server) RegisterName(name string, rcvr interface{}) error {
 		return nil
 	}
 
+	if s.scheme != IPCServer {
+		for _, api := range apiBans {
+			for name, m := range methods {
+				if api == m.method.Func.Pointer() {
+					logger.Trace("The banned API is not registered", "API", name, "scheme", SchemeToString(s.scheme))
+					delete(methods, name)
+				}
+			}
+		}
```

Comment:
Can we use a map instead of double for-loops?

Reply:
Refactored. PTAL again.